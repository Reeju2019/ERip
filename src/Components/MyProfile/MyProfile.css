.orderCard {
  box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.17);
}

.firstCol {
  background-color: #f5f6fa;
  height: 100vh;
}
.secondCol {
  background-color: #fafbfb;
  height: 100vh;
}

.leftButton {
  padding: 0.7vmax;
  padding-left: 2vmax;
  padding-right: 2vmax;
  border: 1px solid blue;
  border-right: none;
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
  font-size: 1.3vmax;
}
.rightButton {
  font-size: 1.3vmax;
  padding: 0.7vmax;
  padding-right: 2vmax;
  padding-left: 2vmax;
  border: 1px solid blue;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.cardContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  height: 80vh;
  overflow-y: scroll;
}
.cardContainer::-webkit-scrollbar {
  overflow-x: hidden;
}

.active {
  background-color: blue;
  color: white;
  cursor: pointer;
}

.profileImg {
  max-width: 7vmax;
  border: 3px solid rgba(128, 128, 128, 0.349);
  border-radius: 50%;
}

.contactNumber {
  font-size: 14px;
}

.update {
  color: rgb(79, 79, 170);
  cursor: pointer;
}

.cancelOrder {
  border: 1px solid red;
  padding: 0.9vmax 3vmax;
  border-radius: 25px;
  text-align: center;
  align-self: center;
  background-color: #fda1aa;
}

.orderImage {
  max-width: 7vmax;
}
.orderId {
  font-size: 1.1vmax;
}
.callSupport {
  font-size: 1.2vmax;
  border: 1px solid blue;
  border-radius: 5px;
  padding: 0.9vmax 2.7vmax;
  color: rgb(9, 9, 156);
  transition: all 1s;
}
.viewDetails {
  font-size: 1.2vmax;
  border: 1px solid blue;
  border-radius: 5px;
  padding: 0.9vmax 2.7vmax;
  color: rgb(9, 9, 156);
  transition: all 1s;
}

.viewDetails:hover,
.callSupport:hover {
  color: white;
  background-color: blue;
}
