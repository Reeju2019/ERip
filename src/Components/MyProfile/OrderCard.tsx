import * as React from 'react'
import './MyProfile.css'
import MyProfileData from '../../Data/MyProfile.mock.json'

interface IOrderCardProps {
  item: {
    order_id: string
    time: string
    device: string
    issue: string
    order_status: string
    image: string
  }
}

const OrderCard: React.FunctionComponent<IOrderCardProps> = ({ item }) => {
  return (
    <>
      <section className='bg-white p-3 orderCard'>
        <div className='d-flex justify-content-between mt-3'>
          <div className='orderId'>
            {MyProfileData.constData.order_id}
            <span>{item.order_id}</span>
          </div>
          <button className='cancelOrder text-danger'>{MyProfileData.constData.cancelOrder}</button>
        </div>
        <div className='d-flex justify-content-between m-3'>
          <div className='col-4'>
            <img src={item.image} className='orderImage' alt={item.issue} />
          </div>
          <div className='col-8'>
            <div className='d-flex flex-row'>
              <i className='fa-solid fa-clock align-self-center mx-2' />
              <span className='fs-5'>{item.time}</span>
            </div>
            <div>
              <i className='fa-solid fa-laptop-code align-self-center mx-2' />
              <span className='fs-5'>{item.device}</span>
            </div>
            <div>
              <i className='fa-solid fa-gears align-self-center mx-2' />
              <span className='fs-5'>{item.issue}</span>
            </div>
          </div>
        </div>
        <p>{item.order_status}</p>
        <div className='d-flex flex-row justify-content-between'>
          <button className='callSupport'>{MyProfileData.constData.callSupport}</button>
          <button className='viewDetails'>{MyProfileData.constData.viewDetails}</button>
        </div>
      </section>
    </>
  )
}

export default OrderCard
