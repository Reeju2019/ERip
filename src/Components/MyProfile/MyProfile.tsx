import React from 'react'
import { useState } from 'react'
import { Col, Container, Row } from 'react-bootstrap'
import './MyProfile.css'
import MyProfileData from '../../Data/MyProfile.mock.json'
import OrderCard from './OrderCard'
import Footer from '../footer/Footer'
import Header from '../Home/Header/Header'

const MyProfile: React.FunctionComponent = () => {
  const [show, setShow] = useState(true)
  const showClicked = () => {
    setShow(!show)
  }
  return (
    <>
      <Header />
      <Container fluid className=''>
        <Row>
          <Col className='col-9 firstCol'>
            <div className='m-3 d-flex justify-content-center  '>
              <button
                disabled={show}
                className={'leftButton ' + (show ? 'active' : '')}
                onClick={showClicked}
              >
                {MyProfileData.constData.ongoing}
              </button>
              <button
                disabled={!show}
                className={'rightButton ' + (show ? '' : 'active')}
                onClick={showClicked}
              >
                {MyProfileData.constData.compleated}
              </button>
            </div>
            <div className='m-1 cardContainer'>
              {MyProfileData &&
                !show &&
                MyProfileData.compleated.map((item, index) => {
                  return (
                    <div key={index} className='col-5 m-3'>
                      <OrderCard item={item} />
                    </div>
                  )
                })}
              {/* {MyProfileData &&
                show &&
                MyProfileData.ongoing.map((item, index) => {
                  return (
                    <div key={index} className='col-5 m-3'>
                      <OrderCard item={item} />
                    </div>
                  )
                })} */}
            </div>
          </Col>
          <Col className='col-3 secondCol'>
            <div className='d-flex flex-column justify-content-center m-4'>
              <div className='profileImg align-self-center m-4'>
                <img src='/static/my_profile/user_prof.png' className='w-100' alt='user' />
              </div>
              <h4 className='d-flex justify-content-center'>{MyProfileData.name}</h4>
              <div className='d-flex flex-row justify-content-center contactNumber'>
                +91
                <span className='mx-1'>{MyProfileData.phoneNumber}</span>
                <i className='fa-solid fa-pencil' />
              </div>
            </div>
            <div className='m-2'>
              <div className='d-flex justify-content-between'>
                <div>{MyProfileData.constData.saved_address}</div>
                <div className='update'>{MyProfileData.constData.update_address}</div>
              </div>
              <hr className='my-2' />
              <div className='d-flex justify-content-between'>
                <div>{MyProfileData.constData.contact_details}</div>
                <div className='update'>{MyProfileData.constData.update}</div>
              </div>
              <hr className='my-2' />
              <div>{MyProfileData.constData.refer}</div>
              <hr className='my-2' />
              <div>{MyProfileData.constData.get_help}</div>
            </div>
          </Col>
        </Row>
      </Container>
      <Footer />
    </>
  )
}

export default MyProfile
