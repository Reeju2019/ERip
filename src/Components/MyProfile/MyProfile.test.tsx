import { fireEvent, getByTitle, render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import { MemoryRouter as Router } from 'react-router-dom'
import MyProfile from './MyProfile'
import React from 'react'

describe('my profile rendering specification', () => {
  // let button: HTMLElement
  test('my profile', () => {
    // const setStateMock = jest.fn()
    // const useStateMock: any = (useState: any) => [useState, setStateMock]
    // jest.spyOn(React, 'useState').mockImplementation(useStateMock)
    // const { getByRole } = render(
      render(
      <Router>
        <MyProfile />
      </Router>,
    )
    // button = getByRole('button', { pressed: true })
    // fireEvent.click(button)

    // expect(setStateMock).toHaveBeenCalledWith(true)
  })
  it('SnapShot testing', () => {
    const component = renderer.create(
      <Router>
        <MyProfile />
      </Router>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render without crashing', () => {
    const div = document.createElement('div')
    render(
      <Router>
        <MyProfile />
      </Router>,
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})
