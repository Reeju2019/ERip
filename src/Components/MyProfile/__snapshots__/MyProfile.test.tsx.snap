// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my profile rendering specification SnapShot testing 1`] = `
Array [
  <div
    className=" py-1 m-0 header-comp container-fluid"
  >
    <div
      className="container"
    >
      <div
        className="Navbar row"
      >
        <div
          className="col-1 align-item-center align-self-center col"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="logo"
              className="col-2 Logo align-self-center"
              src="/static/image/logo_erip.png"
            />
          </a>
        </div>
        <div
          className="modal-search d-flex col-6 col"
        >
          <button
            className="btn modal-btn rounded-0 bg-transparent text-muted fw-bold btn btn-primary"
            data-target="#myModal"
            data-toggle="modal"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-map-marker-alt"
            />
            Bengaluru
            <i
              className="fa-solid fa-caret-down"
            />
          </button>
          <div
            className="search_btn d-flex input-group"
          >
            <div
              className="align-self-center m-3"
            >
              <i
                className="fas fa-search"
              />
            </div>
            <input
              aria-describedby="search-addon"
              aria-label="Search"
              className="form-control rounded-0 border-0 form-control"
              id="search-border"
              placeholder="Search for a services"
              type="search"
            />
          </div>
        </div>
        <div
          className="col-5 align-self-center col"
        >
          <div
            className="modal-search d-flex justify-content-end align-self-center nav"
            onKeyDown={[Function]}
          >
            <a
              className="text-muted fw-bold nav-link"
              data-rr-ui-event-key="/"
              disabled={false}
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
            <a
              className="text-muted fw-bold nav-link"
              data-rr-ui-event-key="/contact"
              disabled={false}
              href="/contact"
              onClick={[Function]}
            >
              Contact Us
            </a>
            <a
              className="d-flex flex-row text-muted fw-bold nav-link"
              data-rr-ui-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <i
                className="fa-regular fa-user"
              />
              Login
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-9 firstCol col"
      >
        <div
          className="m-3 d-flex justify-content-center  "
        >
          <button
            aria-pressed="true"
            className="leftButton active"
            disabled={true}
            onClick={[Function]}
          >
            Ongoing
          </button>
          <button
            className="rightButton "
            disabled={false}
            onClick={[Function]}
          >
            Compleated
          </button>
        </div>
        <div
          className="m-1 cardContainer"
        />
      </div>
      <div
        className="col-3 secondCol col"
      >
        <div
          className="d-flex flex-column justify-content-center m-4"
        >
          <div
            className="profileImg align-self-center m-4"
          >
            <img
              alt="user"
              className="w-100"
              src="/static/my_profile/user_prof.png"
            />
          </div>
          <h4
            className="d-flex justify-content-center"
          >
            Alexa
          </h4>
          <div
            className="d-flex flex-row justify-content-center contactNumber"
          >
            +91
            <span
              className="mx-1"
            >
              9876543210
            </span>
            <i
              className="fa-solid fa-pencil"
            />
          </div>
        </div>
        <div
          className="m-2"
        >
          <div
            className="d-flex justify-content-between"
          >
            <div>
              Saved Address
            </div>
            <div
              className="update"
            >
              Update Address
            </div>
          </div>
          <hr
            className="my-2"
          />
          <div
            className="d-flex justify-content-between"
          >
            <div>
              Contact Details
            </div>
            <div
              className="update"
            >
              Update
            </div>
          </div>
          <hr
            className="my-2"
          />
          <div>
            Refer a Friend
          </div>
          <hr
            className="my-2"
          />
          <div>
            Get Help
          </div>
        </div>
      </div>
    </div>
  </div>,
  <section
    className="footer"
  >
    <div
      className="container-fluid g-0 footer-container"
      style={
        Object {
          "backgroundColor": "#24252a",
          "display": "flex",
          "paddingBottom": "50px",
          "paddingTop": "50px",
        }
      }
    >
      <div
        className="first-footer "
      >
        <img
          alt="white-logo"
          className="erip-logo"
          src="/assets/erip-logo-white.png"
        />
        <p
          className="footer-text"
        >
          Â© Erip 2021 - 2021, All Rights Reserved
        </p>
        <i
          className="fa-brands fa-twitter"
        />
        <i
          className="facebook fab fa-facebook-square"
        />
        <i
          className="Instagram fab fa-instagram"
        />
        <i
          className="youtube fab fa-youtube"
        />
        <i
          className="fa-brands fa-linkedin"
        />
        <i
          className="printerest fab fa-pinterest-p"
        />
      </div>
      <div
        className="col-6"
      >
        <ul
          className="second-footer"
        >
          <li>
            About Us
          </li>
          <li>
            Contact Us
          </li>
          <li>
            Terms of use
          </li>
          <li>
            Customers Support
          </li>
          <li>
            Privacy Policy
          </li>
        </ul>
      </div>
    </div>
  </section>,
]
`;
